### PR TITLE
add support kraken preheat

### PR DESCRIPTION
**Pull request by mmpei**
Signed-off-by: peimingming <peimingming@corp.netease.com>
tested.
kraken endpoint is 59.111.92.135:15005 .

### CODE REVIEW

File: src/pkg/p2p/preheat/provider/kraken.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"strings"
+
+	"github.com/docker/distribution/manifest/schema2"
+	common_models "github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/dao/models"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/provider/auth"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/provider/client"
+	"github.com/goharbor/harbor/src/pkg/registry"
+	"time"
```
Comment (cd1989) [inline src/pkg/p2p/preheat/provider/kraken.go]: Suggest to move `time` up to be grouped with other standard packages.

File: src/pkg/p2p/preheat/provider/kraken.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+func (kd *KrakenDriver) GetHealth() (*DriverStatus, error) {
+	if kd.instance == nil {
+		return nil, errors.New("missing instance metadata")
+	}
+
+	url := fmt.Sprintf("%s%s", strings.TrimSuffix(kd.instance.Endpoint, "/"), krakenHealthPath)
+	_, err := client.DefaultHTTPClient.Get(url, kd.getCred(), nil, nil)
+	if err != nil {
+		// Unhealthy
+		return nil, err
```
Comment (cd1989) [inline src/pkg/p2p/preheat/provider/kraken.go]: Why not `DriverStatusUnHealthy` ?

File: src/pkg/p2p/preheat/provider/kraken.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	// For Dragonfly, no error returned means healthy
```
Comment (cd1989) [inline src/pkg/p2p/preheat/provider/kraken.go]: `For Dragonfly`  --> `For Kraken` ?

Comment (mmpei) [reply src/pkg/p2p/preheat/provider/kraken.go]: ok

File: src/pkg/p2p/preheat/provider/kraken.go (hunk 1 above)
Comment (mmpei) [reply src/pkg/p2p/preheat/provider/kraken.go]: ok

Comment (mmpei) [reply src/pkg/p2p/preheat/provider/kraken.go]: ok

File: src/pkg/p2p/preheat/provider/kraken.go (hunk 2 above)
Comment (mmpei) [reply src/pkg/p2p/preheat/provider/kraken.go]: it seams a uniform format. error will be logged outside.

**Pull request merged**